mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -72,14 +72,16 @@ def stop_app(self):
             raise Error('Failed to stop existing apk. Unexpected output: %s' %
                         out)
 
-    def _is_app_installed(self):
+    def check_app_installed(self):
         """Overrides superclass."""
-        try:
-            out = self._adb.shell(
-                'pm list instrumentation | grep ^instrumentation:%s/' %
-                self.package).decode('utf-8')
-            return bool(out)
-        except adb.AdbError as e:
-            if (e.ret_code == 1) and (not e.stdout) and (not e.stderr):
-                return False
-            raise
+        if not self._adb_grep_wrapper(
+            'pm list package | grep %s' % self.package):
```

Comment:
Can you make the grep more specific? ie grep 'package:%s$'
The snippet's package is usually a superset of the main app's package
eg
com.my.foo
com.my.foo.snippet

So there's opportunities for the grep to detect an app as 'installed' even if it really isn't.